front/lib/api/data_source_view.ts

```diff
@@ -192,6 +211,52 @@ export async function getContentNodesForStaticDataSourceView(
         type: "database",
       }));
 
-    return new Ok(tablesAsContentNodes);
+    return new Ok({
+      nodes: tablesAsContentNodes,
+      total: tablesRes.value.total,
+    });
   }
 }
+
+export async function getContentNodesForDataSourceView(
+  dataSourceView: DataSourceViewResource,
+  params: GetContentNodesForDataSourceViewParams
+): Promise<
+  Result<GetContentNodesForDataSourceViewResult, Error | CoreAPIError>
+> {
+  let contentNodesResult: GetContentNodesForDataSourceViewResult;
+
+  if (dataSourceView.dataSource.connectorId) {
+    const contentNodesRes = await getContentNodesForManagedDataSourceView(
+      dataSourceView,
+      params
+    );
+
+    if (contentNodesRes.isErr()) {
+      return contentNodesRes;
+    }
+
+    contentNodesResult = contentNodesRes.value;
+  } else {
+    const contentNodesRes = await getContentNodesForStaticDataSourceView(
+      dataSourceView,
+      params
+    );
+
+    if (contentNodesRes.isErr()) {
+      return contentNodesRes;
+    }
+
+    contentNodesResult = contentNodesRes.value;
+  }
+
+  const contentNodesInView = filterAndCropContentNodesByView(
+    dataSourceView,
+    contentNodesResult.nodes
+  );
```

Comment:
I don't think it can happen for static data sources - but if filterAndCropContentNodesByView is filtering some items out, the number of items returned may not match the limit ? this will be an issue for managed data sources ?

Reply:
Yes we will see, it should happen since the view filter is applied at runtime, this function also crops the parent ids.